acceptance: TestDockerCLI_test_demo_multitenant failed [failed to create engines]

acceptance.TestDockerCLI_test_demo_multitenant [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/11799691?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/11799691?buildTab=artifacts#/) on master @ [985662236d7bf273b93a7b5e32def8e2d1043640](https://github.com/cockroachdb/cockroach/commits/985662236d7bf273b93a7b5e32def8e2d1043640):


```
=== RUN   TestDockerCLI_test_demo_multitenant
    test_log_scope.go:167: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/_tmp/b38fdc3e3d87d99024418e1ae9af5f4c/logTestDockerCLI_test_demo_multitenant3529712415
    test_log_scope.go:81: use -show-logs to present logs inline
--- FAIL: TestDockerCLI_test_demo_multitenant (66.34s)
=== RUN   TestDockerCLI_test_demo_multitenant/test_demo_multitenant.tcl
    --- FAIL: TestDockerCLI_test_demo_multitenant/test_demo_multitenant.tcl (57.98s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDockerCLI_test_demo_multitenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-31589

acceptance.TestDockerCLI_test_demo_multitenant [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/11808272?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/11808272?buildTab=artifacts#/) on master @ [6cbd07ee6fbfb92706e8cdc8c559960b1bc41663](https://github.com/cockroachdb/cockroach/commits/6cbd07ee6fbfb92706e8cdc8c559960b1bc41663):


```
=== RUN   TestDockerCLI_test_demo_multitenant
    test_log_scope.go:167: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/_tmp/b38fdc3e3d87d99024418e1ae9af5f4c/logTestDockerCLI_test_demo_multitenant204937003
    test_log_scope.go:81: use -show-logs to present logs inline
--- FAIL: TestDockerCLI_test_demo_multitenant (68.63s)
=== RUN   TestDockerCLI_test_demo_multitenant/test_demo_multitenant.tcl
    --- FAIL: TestDockerCLI_test_demo_multitenant/test_demo_multitenant.tcl (58.16s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDockerCLI_test_demo_multitenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

acceptance.TestDockerCLI_test_demo_multitenant [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_Tests_Acceptance/11808865?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_Tests_Acceptance/11808865?buildTab=artifacts#/) on master @ [6cbd07ee6fbfb92706e8cdc8c559960b1bc41663](https://github.com/cockroachdb/cockroach/commits/6cbd07ee6fbfb92706e8cdc8c559960b1bc41663):


```
=== RUN   TestDockerCLI_test_demo_multitenant
    test_log_scope.go:167: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/_tmp/35f27bee8392199acbe6273d72f1cec5/logTestDockerCLI_test_demo_multitenant3707116048
    test_log_scope.go:81: use -show-logs to present logs inline
--- FAIL: TestDockerCLI_test_demo_multitenant (68.08s)
=== RUN   TestDockerCLI_test_demo_multitenant/test_demo_multitenant.tcl
    --- FAIL: TestDockerCLI_test_demo_multitenant/test_demo_multitenant.tcl (58.31s)
```
<p>Parameters: <code>GOFLAGS=-json</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDockerCLI_test_demo_multitenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

acceptance.TestDockerCLI_test_demo_multitenant [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/11853875?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/11853875?buildTab=artifacts#/) on master @ [db645004be34f1b54a2d2aa0661ad78839b00904](https://github.com/cockroachdb/cockroach/commits/db645004be34f1b54a2d2aa0661ad78839b00904):


```
=== RUN   TestDockerCLI_test_demo_multitenant
    test_log_scope.go:167: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/_tmp/b38fdc3e3d87d99024418e1ae9af5f4c/logTestDockerCLI_test_demo_multitenant1888081260
    test_log_scope.go:81: use -show-logs to present logs inline
--- FAIL: TestDockerCLI_test_demo_multitenant (65.27s)
=== RUN   TestDockerCLI_test_demo_multitenant/test_demo_multitenant.tcl
    --- FAIL: TestDockerCLI_test_demo_multitenant/test_demo_multitenant.tcl (57.74s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDockerCLI_test_demo_multitenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/acceptance/generated_cli_test.go
+++ b/pkg/acceptance/generated_cli_test.go
@@ -137,6 +137,13 @@ func TestDockerCLI_test_demo_memory_warning(t *testing.T) {
 	runTestDockerCLI(t, "test_demo_memory_warning", "../cli/interactive_tests/test_demo_memory_warning.tcl")
 }
 
+func TestDockerCLI_test_demo_multitenant(t *testing.T) {
+	s := log.Scope(t)
+	defer s.Close(t)
+
+	runTestDockerCLI(t, "test_demo_multitenant", "../cli/interactive_tests/test_demo_multitenant.tcl")
+}
+
 func TestDockerCLI_test_demo_networking(t *testing.T) {
 	s := log.Scope(t)
 	defer s.Close(t)

--- a/pkg/cli/democluster/demo_cluster.go
+++ b/pkg/cli/democluster/demo_cluster.go
@@ -1060,6 +1060,13 @@ func (c *transientCluster) DrainAndShutdown(ctx context.Context, nodeID int32) e
 	if err := c.drainAndShutdown(ctx, c.servers[serverIdx].adminClient); err != nil {
 		return err
 	}
+
+	select {
+	case <-c.servers[serverIdx].Stopper().IsStopped():
+	case <-time.After(10 * time.Second):
+		return errors.Errorf("server stopper not stopped after 10 seconds")
+	}
+
 	c.servers[serverIdx].TestServerInterface = nil
 	c.servers[serverIdx].adminClient = nil
 	if c.demoCtx.Multitenant {

--- a/pkg/cli/interactive_tests/test_demo_multitenant.tcl
+++ b/pkg/cli/interactive_tests/test_demo_multitenant.tcl
@@ -1,9 +1,5 @@
 #! /usr/bin/env expect -f
 
-# This test is skipped -- its filename lets it hide from the selector in
-# TestDockerCLI. Unskip it by renaming after fixing
-# https://github.com/cockroachdb/cockroach/issues/110748.
-
 source [file join [file dirname $argv0] common.tcl]
 
 spawn $argv demo --no-line-editor --empty --nodes 3 --multitenant --log-dir=logs

